### PR TITLE
Updated Plugin

### PR DESCRIPTION
**Pull request by mikeynap**
### Change log
Updated my w0nderful plugin client.
Added some Input Parsing methods.

### CODE REVIEW

File: plugin/client/plugin.go [hunk 1]
```diff
@@ -20,7 +20,6 @@ const (
 	PluginArgUndefined           = "undefined"
 )
 
-//GetArgs return the plugin Argument, it must be from the specifed consts to be known.
 func GetArg() PluginArg {
```
Comment (meganokeefe) [inline plugin/client/plugin.go]: Can you bring back the function header comments that were in `plugin.go` before? Now that we have godoc https://godoc.org/github.com/sigma-dev/sigma

File: plugin/client/plugin.go [hunk 2]
```diff
@@ -46,60 +44,87 @@ func GetInputs() (map[string][]string, error) {
 	return m, nil
 }
 
-//Outputs defines a map and savePath for all outputs
-type Outputs struct {
-	m        map[string][]string
-	savePath string
+func LenEqual(in ...[]string) bool {
```
Comment (meganokeefe) [inline plugin/client/plugin.go]: Comment these helpers and their purpose

File: task/docker.go [hunk 3]
```diff
@@ -144,7 +145,7 @@ func (d *DockerTasker) Remove(so *model.ServiceOperation) error {
 	if strings.ContainsAny(so.Service.Id, "./~") {
 		return ErrMalformedID
 	}
-	defer os.RemoveAll(path.Join(d.saveDir, so.Service.Id))
+	//defer os.RemoveAll(path.Join(d.saveDir, so.Service.Id))
```
Comment (stephenrlouie) [inline task/docker.go]: ?

File: plugin/client/plugin_test.go [hunk 4]
```diff
@@ -10,18 +10,22 @@ import (
 )
 
 func TestPlugin(t *testing.T) {
```
Comment (stephenrlouie) [inline plugin/client/plugin_test.go]: can we get a little more coverage?

File: plugin/client/plugin.go (hunk 2 above)
Comment (meganokeefe) [inline plugin/client/plugin.go]: also I don't see `LenEqual()` used anywhere in the project.

File: plugin/client/plugin.go [hunk 5]
```diff
@@ -46,60 +44,87 @@ func GetInputs() (map[string][]string, error) {
...
+	}
+	return nil
 }
 
-//NewOutputs creates a new Output instance
-func NewOutputs(savePath string) Outputs {
-	return Outputs{
-		make(map[string][]string),
-		savePath,
+func ParseInputAsString(key string, in map[string][]string, errs *[]error) string {
```
Comment (meganokeefe) [inline plugin/client/plugin.go]: Same with `ParseInputAsString()`, where is the function going to be used? Doesn't seem to be used right now

Comment (stephenrlouie) [inline plugin/client/plugin.go]: a plugin would use these

File: task/docker.go (hunk 3 above)
Comment (mikeynap) [inline task/docker.go]: Oops

File: plugin/client/plugin.go (hunk 2 above)
Comment (mikeynap) [inline plugin/client/plugin.go]: It's for plugins.

**Comment by meganokeefe** [comment]
Ya got merge conflicts. But once you fix, I'm good 2 merge this.

File: plugin/client/plugin.go [hunk 6]
```diff
@@ -32,74 +31,83 @@ func GetArg() PluginArg {
...
-}
-
-//NewOutputs creates a new Output instance
-func NewOutputs(savePath string) Outputs {
-	return Outputs{
-		make(map[string][]string),
-		savePath,
+// LenEqual is a helper function for inputs. Tests if len(in[0]) == len(in[1]) == ... == len(in[n])
+func LenEqual(in ...[]string) bool {
+	for i := 0; i < len(in)-1; i++ {
```
Comment (meganokeefe) [inline plugin/client/plugin.go]: you are not using this function anywhere, that I can see. 

is this just here as setup for future changes that are gonna use this? what is this functionally for, i am still confused

**Pull request merged**